Slim/Extras/Middleware/FeatherBB.php

```diff
@@ -1,36 +1,132 @@
 <?php
 /**
- * FeatherBB
  *
- *
- * USAGE
+ * Copyright (C) 2015 FeatherBB
+ * based on code by (C) 2008-2012 FluxBB
+ * and Rickard Andersson (C) 2002-2008 PunBB
+ * License: http://www.gnu.org/licenses/gpl.html GPL version 2 or higher
  *
  * $app = new \Slim\Slim();
- * $app->add(new \Slim\Extras\Middleware\FeatherBB());
+ * $app->add(new \Slim\Extras\Middleware\FeatherBB($params));
  *
  */
+
 namespace Slim\Extras\Middleware;
+use DB;
 
 class FeatherBB extends \Slim\Middleware
 {
-	protected $settings = array();
+    protected $data = array();
 
-	public function __construct()
+	public function __construct(array $user_forum_settings = array())
 	{
+        // Define forum env constants
+        $this->data['forum_env'] = array(
```

Comment:
No, it only defines the constants which were originally defined in common.php. Keep in mind that the middleware is called before the Slim application in the framework stack